packages/web-components/src/components/ic-dialog/ic-dialog.tsx

```diff
@@ -0,0 +1,371 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Prop,
+  State,
+  Method,
+  Listen,
+  h,
+} from "@stencil/core";
+import closeIcon from "../../assets/close-icon.svg";
+import { isSlotUsed } from "../../utils/helpers";
+
+@Component({
+  tag: "ic-dialog",
+  styleUrl: "ic-dialog.css",
+  shadow: true,
+})
+export class Dialog {
+  /**
+   * @slot - Content will be placed in the middle of the dialog, between the heading and the controls.
+   * @slot dialog-controls - Content will be place at the bottom of the dialog.
+   */
+  @Element() el: HTMLIcDialogElement;
+
+  /**
+   * Sets the maximum and minimum height and width for the dialog.
```

Comment:
Do we need to add something to explain that for xs breakpoint, these have no effect and the dialog is always full screen? Or is that something that should be mentioned in the guidance?

Reply:
I agree its definitely worth mentioning, not sure whether it would be better as guidance or mentioned somewhere within the component.